### PR TITLE
Light refactoring

### PR DESCRIPTION
**Pull request by maxulysse**
The linter is failing, but that is a [bug](https://github.com/nf-core/tools/issues/2833) in the linter which has been solved in the dev-version of the tool.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by asp8200** [comment]
So far this PR seems like it is adding some stuff and not refactoring 😄  Anyways, linter says

"Param modules_testdata_base_path from nextflow config not found in nextflow_schema.json"

**Review by FriederikeHanssen — DISMISSED** [review]
lgtm 🚀

**Review by kenibrewer — DISMISSED** [review]
LGTM too

**Review by kenibrewer — DISMISSED** [review]

**Comment by kenibrewer** [comment]
There's a nf-core linting failure that needs to be addressed before review:

> [nextflow_config](https://nf-co.re/tools/docs/2.13.1/pipeline_lint_tests/nextflow_config) - Config default value incorrect: params.aligner is set as bwa-mem in nextflow_schema.json but is null in nextflow.config.

File: main.nf [hunk 1]
```diff
@@ -354,6 +352,25 @@ workflow {
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    FUNCTIONS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+//
+// Get attribute from genome config file e.g. fasta
+//
+
+def getGenomeAttribute(attribute) {
```
Comment (asp8200) [inline main.nf:365]: but then we also got `getGenomeAttribute` defined [here](https://github.com/maxulysse/nf-core_sarek/blob/cbb2a62180d6a01f894acfd4ade04ae8735da22f/subworkflows/local/utils_nfcore_sarek_pipeline/main.nf#L258).  That seems odd 🤔

**Comment by asp8200** [comment]
In [4575cba](https://github.com/nf-core/sarek/pull/1422/commits/4575cbab5415212f62e2e929343b04b3859f2d58), then full-test is still not running on awsbatch :-/  I still get:
```
ERROR ~ Error executing process > 'NFCORE_SAREK:PREPARE_INTERVALS:CREATE_INTERVALS_BED (wgs_calling_regions.hg38.bed)'

Caused by:
  The user value contains invalid characters. Enter a value that matches the pattern ^([a-z0-9_][a-z0-9_-]{0,30})$

Command executed:

  awk -vFS="    " '{
      t = $5  # runtime estimate
      if (t == "") {
          # no runtime estimate in this row, assume default value
          t = ($3 - $2) / 200000
      }
      if (name == "" || (chunk > 600 && (chunk + t) > longest * 1.05)) {
          # start a new chunk
          name = sprintf("%s_%d-%d.bed", $1, $2+1, $3)
          chunk = 0
          longest = 0
      }
      if (t > longest)
          longest = t
      chunk += t
      print $0 > name
  }' wgs_calling_regions.hg38.bed

  cat <<-END_VERSIONS > versions.yml
  "NFCORE_SAREK:PREPARE_INTERVALS:CREATE_INTERVALS_BED":
      gawk: $(awk -Wversion | sed '1!d; s/.*Awk //; s/,.*//')
  END_VERSIONS

Command exit status:
  -

Command output:
  (empty)
```
(The full-tests also fails no the dev-branch, but due to other bugs which have now been fixed in this PR.)

The "small" test-profile also fail on  [4575cba](https://github.com/nf-core/sarek/pull/1422/commits/4575cbab5415212f62e2e929343b04b3859f2d58) with a similar error but from another process:
```
ERROR ~ Error executing process > 'NFCORE_SAREK:PREPARE_INTERVALS:TABIX_BGZIPTABIX_INTERVAL_COMBINED (wgs_calling_regions.hg38)'

Caused by:
  The user value contains invalid characters. Enter a value that matches the pattern ^([a-z0-9_][a-z0-9_-]{0,30})$

Command executed:

  bgzip  --threads 1 -c  wgs_calling_regions.hg38.bed > wgs_calling_regions.hg38.bed.gz
  tabix  wgs_calling_regions.hg38.bed.gz

  cat <<-END_VERSIONS > versions.yml
  "NFCORE_SAREK:PREPARE_INTERVALS:TABIX_BGZIPTABIX_INTERVAL_COMBINED":
      tabix: $(echo $(tabix -h 2>&1) | sed 's/^.*Version: //; s/ .*$//')
  END_VERSIONS

Command exit status:
  -

Command output:
  (empty)
```

**Comment by kenibrewer** [comment]
@asp8200 Could the problem maybe be happening when we try to [assign a tag](https://github.com/maxulysse/nf-core_sarek/blob/4575cbab5415212f62e2e929343b04b3859f2d58/modules/local/create_intervals_bed/main.nf#L2) to the job? Maybe AWS batch only supports tags of a certain length, or the files for the AWS Batch dataset are too long / contain illegal characters.

File: nextflow.config [hunk 2]
```diff
@@ -134,7 +135,7 @@ params {
     // Schema validation default options
     validationFailUnrecognisedParams = false
     validationLenientMode            = true
-    validationSchemaIgnoreParams     = 'cf_ploidy,genomes,igenomes_base'
+    validationSchemaIgnoreParams     = 'input_restart,cf_ploidy,genomes,igenomes_base,ascat_alleles,ascat_genome,ascat_loci,ascat_loci_gc,ascat_loci_rt,bwa,bwamem2,cf_chrom_len,chr_dir,dbsnp,dbsnp_tbi,dbsnp_vqsr,dict,dragmap,fasta,fasta_fai,germline_resource,germline_resource_tbi,intervals,known_indels,known_indels_tbi,known_indels_vqsr,known_snps,known_snps_tbi,known_snps_vqsr,mappability,ngscheckmate_bed,pon,pon_tbi,sentieon_dnascope_model,snpeff_db,snpeff_genome,vep_cache_version,vep_genome,vep_species'
```
Comment (asp8200) [inline nextflow.config:138]: Seems strange that `validationSchemaIgnoreParams` should ignore so many parameters? 🤔  Is that just some kind of temporary fix?

**Review by asp8200 — APPROVED** [review]
Not sure why stuff is being moved around, but if it works and isn't more confusing than before, then OK. By the way, the error reported from the linter can be ignored. It is a know bug in the linter.

**Comment by kenibrewer** [comment]
@maxulysse What still remains to be done on this PR? This is a pretty substantial bit of refactoring, and I think it would be good to get this merged back into dev sooner rather than later.

Also, as general practice for nf-core development,  is it acceptable for major rewrites like this one to merge a mostly correct set of changes into dev and then address the remaining failures as PRs to that branch?

**Review by asp8200 — APPROVED** [review]
LGTM.  However, we still need to fix the runOptions/containerOptions-issue so that the pipeline can run over awsbatch. We do have a fix for that but it isn't great:

https://nextflow.slack.com/archives/C02T98A23U7/p1711040612294759?thread_ts=1711013350.318419&cid=C02T98A23U7

https://github.com/nf-core/sarek/pull/1430/files

**Comment by maxulysse** [comment]
> LGTM. However, we still need to fix the runOptions/containerOptions-issue so that the pipeline can run over awsbatch. We do have a fix for that but it isn't great:
> 
> https://nextflow.slack.com/archives/C02T98A23U7/p1711040612294759?thread_ts=1711013350.318419&cid=C02T98A23U7
> 
> https://github.com/nf-core/sarek/pull/1430/files

yeah, but that's another issue

**Pull request merged**